src/Client/Adapter/Curl.php

```diff
@@ -203,9 +203,9 @@ public function connect($host, $port = 80, $secure = false)
         }
 
         if (isset($this->config['connecttimeout'])) {
-            $connectTimeout = $this->config['connecttimeout'];
+            $connectTimeout = (int) $this->config['connecttimeout'];
```

Comment:
I would prefer the following approach:

- Grab the config value.
- If it is not an integer or not a numeric string, throw an `InvalidArgumentException`.
- Cast to integer.

That approach ensures strings like `timeout` do not get cast to `0`, which is obviously incorrect.

Reply:
do we need to check `$this->config['timeout']` and $this->config['connecttimeout'] in `setOptions()` function ? and change it at Socket adapter as well?

Reply:
I've updated with integer and numeric check, applied to socket adapter as well.